### DISCUSSION TITLE
Which is better for making state readonly: a getter or using readonly()?

### DISCUSSION THREAD

**Discussion by yoshi-pi** [vuejs/pinia#2479] [Polls]
When making a state readonly, you have the option to use either a getter or the readonly() API. Which approach do you prefer?

```javascript
import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
defineStore('user', () => {
  const _user = ref(null)
  const user = computed(() => _user) // getter
  return { user, _user }
})
```

```javascript
import { defineStore } from 'pinia'
import { ref, computed, readonly } from 'vue'
defineStore('user', () => {
  const user = ref(null)
  return { user: readonly(user) } // readonly()
})
```

**Comment by posva** [comment]
Note you **must return** `_user` too because it's state (https://pinia.vuejs.org/core-concepts/#Setup-Stores). There are better ways to declare private state. I have some lessons and exercises in https://masteringpinia.com/ that I need to port back to the docs so they are accessible to everybody

**Reply by yoshi-pi** [reply]
Thank you for your reply. I've edited it! I found an explanation about private state on [this blog](https://masteringpinia.com/blog/my-top-5-tips-for-using-pinia). By the way, if all the states aren't returned, what kinds of issues or bugs might arise?

**Reply by posva** [reply]
ah right! I forgot I published that article too

**Reply by posva** [reply]
It will break in multiple scenarios: plugins, ssr, devtools
